.github/workflows/tests.yml

```diff
@@ -19,10 +19,20 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest]
+        os: ['ubuntu-latest', 'macos-latest']
         python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
         cratedb-version: ['5.1.1']
         sqla-version: ['1.3.24', '1.4.44']
+        # To save resources, only use the most recent Python version on macOS.
+        exclude:
+          - os: 'macos-latest'
+            python-version: '3.7'
+          - os: 'macos-latest'
+            python-version: '3.8'
+          - os: 'macos-latest'
+            python-version: '3.9'
+          - os: 'macos-latest'
+            python-version: '3.10'
```

Comment:
Should we keep 3.10 as well? both 3.10 and 3.11 are considered stable https://devguide.python.org/versions/#versions

Reply:
Thanks for your attention. I also thought about it, but I think it will be fine. Speeding up the test suite with the many branches currently going on resonates much better with me. When there is demand, I will be happy to turn it back on afterwards.